mesonbuild/compilers/mixins/gnu.py

```diff
@@ -539,6 +540,59 @@ def get_preprocess_to_file_args(self) -> T.List[str]:
         lang = gnu_lang_map.get(self.language, 'assembler-with-cpp')
         return self.get_preprocess_only_args() + [f'-x{lang}']
 
+    def make_arguments_abstract(self, args: T.List[str]) -> T.List[arguments.Argument]:
+        ret: T.List[arguments.Argument] = []
+
+        for arg in args:
+            if arg.startswith('-D'):
+                v: T.Optional[str]
+                arg = arg.removeprefix('-D')
+                if '=' in arg:
+                    k, v = arg.split('=')
+                else:
+                    k, v = arg, None
+                ret.append(arguments.Define(k, v))
```

Comment:
Defines can be spelled on the command line as `-D KEY=value` (with the space after the `-D` switch) meaning that they are seen as two arguments, not as one. Thus, I think that arguments parsing needs to be more sophisticated. Maybe using something like `argparse` may make the code simpler.